server/historian/packages/historian-base/src/routes/utils.ts

```diff
@@ -118,18 +123,44 @@ export async function createGitService(createArgs: createGitServiceArgs): Promis
 	const customData: ITenantCustomDataExternal = details.customData;
 	const writeToExternalStorage = !!customData?.externalStorageData;
 	const storageUrl = config.get("storageUrl") as string | undefined;
+	const ignoreEphemeralFlag: boolean = config.get("ignoreEphemeralFlag");
 	const maxCacheableSummarySize: number =
 		config.get("restGitService:maxCacheableSummarySize") ?? 1_000_000_000; // default: 1gb
 
-	let isEphemeral = cache ? isEphemeralContainer : false;
+	let isEphemeral: boolean = isEphemeralContainer;
 	if (!ignoreEphemeralFlag) {
-		if (isEphemeralContainer !== undefined) {
-			await cache?.set(`isEphemeral:${documentId}`, isEphemeralContainer);
+		const isEphemeralKey: string = `isEphemeralContainer:${documentId}`;
+		// eslint-disable-next-line eqeqeq
+		if (isEphemeral != undefined) {
+			// If an isEphemeral flag was passed in, cache it
+			Lumberjack.info(
+				`Setting ${isEphemeralKey} to ${isEphemeral}`,
+				getLumberBaseProperties(tenantId, documentId),
+			);
+			await cache?.set(isEphemeralKey, isEphemeral);
 		} else {
-			isEphemeral = await cache?.get(`isEphemeral:${documentId}`);
-			// Todo: If isEphemeral is still undefined fetch the value from database
+			isEphemeral = await cache?.get(isEphemeralKey);
+			// eslint-disable-next-line eqeqeq
+			if (isEphemeral == undefined) {
+				// If isEphemeral was not in the cache, fetch the value from database
+				try {
+					const staticProps: IDocumentStaticProperties =
+						await documentManager.readStaticProperties(tenantId, documentId);
+					isEphemeral = staticProps?.isEphemeralContainer ?? false;
+					await cache?.set(isEphemeralKey, isEphemeral); // Cache the value from the static data
+				} catch (e) {
+					isEphemeral = false;
+				}
+			}
 		}
+	} else {
+		isEphemeral = false;
 	}
+	Lumberjack.info(
+		`Document is ephemeral? ${isEphemeral}`,
+		getLumberBaseProperties(tenantId, documentId),
```

Comment:
I think the first param in getLumberBaseProperties should be docId and tenantId is the 2nd one

Reply:
For the other Lumberhack.info log as well

Reply:
Ah good catch